wsmaster/che-core-api-logger-shared/pom.xml

```diff
@@ -0,0 +1,115 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright (c) 2012-2018 Red Hat, Inc.
+    All rights reserved. This program and the accompanying materials
+    are made available under the terms of the Eclipse Public License v1.0
+    which accompanies this distribution, and is available at
+    http://www.eclipse.org/legal/epl-v10.html
+
+    Contributors:
+      Red Hat, Inc. - initial API and implementation
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>che-master-parent</artifactId>
+        <groupId>org.eclipse.che.core</groupId>
+        <version>6.3.0-SNAPSHOT</version>
+    </parent>
+    <artifactId>che-core-api-logger-shared</artifactId>
+    <packaging>jar</packaging>
+    <name>Che Core :: API :: Logger :: Shared</name>
+    <properties>
+        <dto-generator-out-directory>${project.build.directory}/generated-sources/dto/</dto-generator-out-directory>
+    </properties>
+    <dependencies>
+        <dependency>
+            <groupId>com.google.code.gson</groupId>
+            <artifactId>gson</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.eclipse.che.core</groupId>
+            <artifactId>che-core-api-dto</artifactId>
+        </dependency>
+    </dependencies>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.eclipse.che.core</groupId>
+                <artifactId>che-core-api-dto-maven-plugin</artifactId>
+                <version>${project.version}</version>
+                <executions>
+                    <execution>
+                        <id>generate-server-dto</id>
+                        <phase>process-sources</phase>
+                        <goals>
+                            <goal>generate</goal>
+                        </goals>
+                        <configuration>
+                            <dtoPackages>
+                                <package>org.eclipse.che.api.logger.shared.dto</package>
+                            </dtoPackages>
+                            <outputDirectory>${dto-generator-out-directory}</outputDirectory>
+                            <genClassName>org.eclipse.che.api.logger.server.dto.DtoServerImpls</genClassName>
+                            <impl>server</impl>
```

Comment:
I'd say that it is better to generate DTO server implementations in server module. Because a shared module is supposed to contain only shared things which are compatible with dependent modules.
So GWT module will get classes that can't be used there.
WDYT? Does it make sense?

Reply:
I copied content from workspace-shared module so maybe if I'm doing it wrong there is an issue with this module as well ?

it's up to you, let me know what is the better approach.
maybe client or server should be additional classifier artifacts

main artifacts = interfaces
client classifier = for gwt
server classifier = for server

Reply:
I like the idea with classifiers. 

> Is an issue with this module as well 

I think yes.

I think now we can leave it as it is. And later rework it in all places.
@skabashnyuk WDYT?

Reply:
I think we can revise it later.